crates/re_viewer/src/app.rs

```diff
@@ -814,32 +814,24 @@ impl App {
     fn handle_dropping_files(&mut self, store_hub: &mut StoreHub, egui_ctx: &egui::Context) {
         preview_files_being_dropped(egui_ctx);
 
-        // Collect dropped files:
-        if egui_ctx.input(|i| i.raw.dropped_files.len()) > 2 {
-            rfd::MessageDialog::new()
-                .set_level(rfd::MessageLevel::Error)
-                .set_description("Can only load one file at a time")
-                .show();
-        }
-        if let Some(file) = egui_ctx.input(|i| i.raw.dropped_files.first().cloned()) {
-            if let Some(bytes) = &file.bytes {
-                let mut bytes: &[u8] = &(*bytes)[..];
-                if let Some(rrd) = crate::loading::load_file_contents(&file.name, &mut bytes) {
-                    self.on_rrd_loaded(store_hub, rrd);
-
-                    #[allow(clippy::needless_return)] // false positive on wasm32
-                    return;
+        egui_ctx.input(|i| {
+            for file in &i.raw.dropped_files {
```

Comment:
it would be nice to use `input_mut` and the `std::mem::take` the `dropped_files` in order to not hold the mutex on `egui_ctx.input` for too long, i.e.:

```rust
let dropped_files = egui_ctx.input_mut(|i| std::mem::take(i.raw.dropped_files));
for file in dropped_files {
```

Reply:
Niiiice. I wasn't coming up with that one by myself... (yet)